website/src/content/development/notifications.mdx

```diff
@@ -0,0 +1,387 @@
+
+<Info>
+
+In the following code snippet we take advantage of the fact that a `Channel`'s `key` value [must be unique](https://docs.commercetools.com/api/projects/channels#channel).
+A request to update `Channel`'s `key` to a duplicated value results in error with `DuplicateField` error code.
+
+More information about API errors can be found on the [API errors page](https://docs.commercetools.com/api/errors).
+
+</Info>
+
+Having that in mind, we might for instance want to notify the user directly in the `<TextField>` error message in case they try to submit a value that has already been used.
+
+For this purpose, we recommend creating an error transforming function:
+
+```js title="transform-errors.js"
+import omitEmpty from 'omit-empty-es';
+
+const DUPLICATE_FIELD_ERROR_CODE = 'DuplicateField'; // A particular error code returned by API that we wish to map to form field validation error
+
+export const transformErrors = (graphQlErrors) => {
+  const errorsToMap = Array.isArray(graphQlErrors)
+    ? graphQlErrors
+    : [graphQlErrors];
+
+  const { formErrors, unmappedErrors } = errorsToMap.reduce(
+    (transformedErrors, graphQlError) => {
+      const errorCode = graphQlError?.extensions?.code ?? graphQlError.code;
+      const fieldName = graphQlError?.extensions?.field ?? graphQlError.field;
+
+      if (errorCode === DUPLICATE_FIELD_ERROR_CODE) {
+        transformedErrors.formErrors[fieldName] = { duplicate: true };
+      } else {
+        transformedErrors.unmappedErrors.push(graphQlError);
+      }
+      return transformedErrors;
+    },
+    {
+      formErrors: {}, // will be mappped to form field error messages
+      unmappedErrors: [], // will result in dispatching `showApiErrorNotification`
+    }
+  );
+
+  return {
+    formErrors: omitEmpty(formErrors),
+    unmappedErrors,
+  };
+};
+```
+
+And then use it in the `<ChannelsDetails>` component:
+```js title="channels-details.js" highlightLines="2,7,12-18"
+// ...
+import { transformErrors } from './transform-errors';
+
+const ChannelsDetails = (props) => {
+  // ...
+  const handleSubmit = useCallback(
+    async (formValues, formikHelpers) => {
+      const data = formValuesToDoc(formValues);
+      try {
+        // ...
+      } catch (graphQLErrors) {
+        const transformedErrors = transformErrors(graphQLErrors);
+        if (transformedErrors.unmappedErrors.length > 0)
+          showApiErrorNotification({
+            errors: transformedErrors.unmappedErrors,
+          });
+
+        formikHelpers.setErrors(transformedErrors.formErrors);
+      }
+    },
+    [ /* ... */ ]
+  );
+  // ...
+};
+```
+
+Now `DuplicateField` errors related to the `key` field will be transformed to form validation errors and all other errors to API error notifications.
+
+### REST
+
+For HTTP requests sent to REST API we may follow the same pattern as we used for [GraphQL](#graphql):
```

Comment:
```suggestion
For HTTP requests sent to REST API we may follow the same pattern as we used for [GraphQL](#graphql) but use the special hook `showApiErrorNotification`:
```

Reply:
This one I don't fully get, we use `showApiErrorNotification `, right? 🤔